src/librustc_mir/transform/const_debuginfo.rs

```diff
@@ -0,0 +1,86 @@
+//! Finds locals which are assigned once to a const and unused except for debuginfo and converts
+//! their debuginfo to use the const directly, allowing the local to be removed.
+
+use rustc_middle::{ty::{Const, TyCtxt}, mir::{Location, visit::{PlaceContext, Visitor}, Local, Body, StatementKind, Rvalue, Operand, VarDebugInfoContents}};
```

Comment:
:joy:

Reply:
RA likes to do this and I don't actually mind it when it's properly formatted.